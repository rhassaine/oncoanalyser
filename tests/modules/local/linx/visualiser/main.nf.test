nextflow_process {
    name "Test Process LINX_VISUALISER"
    script "modules/local/linx/visualiser/main.nf"
    process "LINX_VISUALISER"
    config "tests/nextflow.config"

    test("LINX_VISUALISER with default parameters") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor'],
                    file('/home/rhassaine/container_test_output/subject_a/linx/somatic_annotations/')
                ]
                input[1] = 38
                input[2] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/common/ensembl_data/", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.plots != null },
                { assert process.out.plots.size() > 0 },
                { assert snapshot(process.out).match() }
            )

            if (process.success) {
                def plots_dir = process.out.plots[0][1]
                def sample_id = process.out.plots[0][0].sample_id

                // Check that the 'all' and 'reportable' directories exist
                assertAll(
                    { assert path(plots_dir).resolve("all").exists() },
                    { assert path(plots_dir).resolve("reportable").exists() }
                )

                // Check the content of 'all' directory
                def all_plots = path(plots_dir).resolve("all").list()
                if (all_plots.size() > 0) {
                    assertAll(
                        { assert all_plots.every { it.fileName.toString().startsWith(sample_id) && it.fileName.toString().endsWith(".png") } },
                        { assert all_plots.any { it.fileName.toString().contains(".chr") || it.fileName.toString().contains(".cluster-") } }
                    )
                } else {
                    println "No plots generated in 'all' directory"
                }

                // Check the content of 'reportable' directory
                def reportable_plots = path(plots_dir).resolve("reportable").list()
                if (reportable_plots.size() > 0) {
                    assertAll(
                        { assert reportable_plots.every { it.fileName.toString().startsWith(sample_id) && it.fileName.toString().endsWith(".png") } },
                        { assert reportable_plots.every { it.fileName.toString().contains(".cluster-") } }
                    )
                } else {
                    println "No plots generated in 'reportable' directory"
                }

                // Check versions
                assertAll(
                    { assert process.out.versions != null },
                    { assert process.out.versions.size() > 0 },
                    { assert path(process.out.versions[0]).exists() }
                )
            }
        }
    }

    test("LINX_VISUALISER should fail when essential input is missing") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = []  // Missing input
                """
            }
        }

        then {
            assertAll(
                { assert !process.success },
                { assert process.exitStatus != 0 }
            )
        }
    }
}