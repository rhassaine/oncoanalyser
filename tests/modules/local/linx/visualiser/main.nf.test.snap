{
    "LINX_VISUALISER with default parameters": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            [
                                "subject_a.tumor.chr11.007.png:md5,6de04c56c1088ce2234bb5c9205a5956",
                                "subject_a.tumor.chr19.000.png:md5,16150521da7abc854d281dd8a78e0f3e",
                                "subject_a.tumor.chr9.007.png:md5,dfbf4080d61c632112df960cf24e4d08",
                                "subject_a.tumor.cluster-0.sv2.003.png:md5,87ff4eeea0f6d97f69cb4bfa74adf55a",
                                "subject_a.tumor.cluster-1.sv2.007.png:md5,c9ad060f12bf078e9ab1745ce3beb505"
                            ],
                            [
                                "subject_a.tumor.cluster-1.sv2.007.png:md5,c9ad060f12bf078e9ab1745ce3beb505"
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,7cc2e10dbc189797869850ae60178730"
                ],
                "plots": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            [
                                "subject_a.tumor.chr11.007.png:md5,6de04c56c1088ce2234bb5c9205a5956",
                                "subject_a.tumor.chr19.000.png:md5,16150521da7abc854d281dd8a78e0f3e",
                                "subject_a.tumor.chr9.007.png:md5,dfbf4080d61c632112df960cf24e4d08",
                                "subject_a.tumor.cluster-0.sv2.003.png:md5,87ff4eeea0f6d97f69cb4bfa74adf55a",
                                "subject_a.tumor.cluster-1.sv2.007.png:md5,c9ad060f12bf078e9ab1745ce3beb505"
                            ],
                            [
                                "subject_a.tumor.cluster-1.sv2.007.png:md5,c9ad060f12bf078e9ab1745ce3beb505"
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7cc2e10dbc189797869850ae60178730"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T05:48:54.926643986"
    }
}